### PR TITLE
provides an example on how to register custom admission plugin.

### PR DESCRIPTION
**Pull request by p0lyn0mial**
this PR adds an example that highlights relevant parts of source code from sample apiserver that deals with admission registration.

close #47868

### CODE REVIEW

File: contributors/design-proposals/apiserver-build-in-admission-plugins.md [hunk 1]
```diff
@@ -53,6 +53,31 @@ following series of steps:
...
  3. Reference the plugin in the admission chain
 
-**TODO**(p0lyn0mial): There is also a [sample apiserver](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-apiserver/main.go) to demonstrate the usage of the generic API library. 
-After implementation sample could would be placed there - copy & paste it here and include a reference.
+## An example
+The sample apiserver provides an example admission plugin that makes meaningful use of the "standard" plugin initializer. 
+The admission plugin ensures that a resource name is not on the list of banned names.
+The source code of the plugin can be found [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission.go). 
+
+Having the plugin the next step is the regitration. [AddmissionOptions](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/options/admission.go)
```
Comment (deads2k) [inline contributors/design-proposals/apiserver-build-in-admission-plugins.md]: typo Admission

File: contributors/design-proposals/apiserver-build-in-admission-plugins.md [hunk 2]
```diff
@@ -53,6 +53,31 @@ following series of steps:
...
+provides two important things. Firstly it exposes [a register](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/options/admission.go#L43) 
+under which all addmission plugins are registered. In fact that's exactly what the [Register](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission.go#L33) 
+method does from our example addmision plugin. It accepts a global registry as a parameter and then simply registers itself in that registry.
+Secondly it adds admission chain to the server configuration via [ApplyTo](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/options/admission.go#L66) method.
+The method accepts optional parameter in the form of `pluginInitalizers`. This is usefull when admission plugins need custom configuration that is not provided by the generic initializer.
```
Comment (deads2k) [inline contributors/design-proposals/apiserver-build-in-admission-plugins.md]: typo useful

File: contributors/design-proposals/apiserver-build-in-admission-plugins.md [hunk 3]
```diff
@@ -53,6 +53,28 @@ following series of steps:
  2. Register the plugin 
  3. Reference the plugin in the admission chain
 
-**TODO**(p0lyn0mial): There is also a [sample apiserver](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-apiserver/main.go) to demonstrate the usage of the generic API library. 
-After implementation sample could would be placed there - copy & paste it here and include a reference.
+## An example
+The sample apiserver provides an example admission plugin that makes meaningful use of the "standard" plugin initializer. 
+The admission plugin ensures that a resource name is not on the list of banned names.
+The source code of the plugin can be found [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission.go).
```
Comment (sttts) [inline contributors/design-proposals/apiserver-build-in-admission-plugins.md]: those links tend to get invalid. Maybe choose a fixed version?

File: contributors/design-proposals/apiserver-build-in-admission-plugins.md [hunk 4]
```diff
@@ -53,6 +53,28 @@ following series of steps:
...
  3. Reference the plugin in the admission chain
 
-**TODO**(p0lyn0mial): There is also a [sample apiserver](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-apiserver/main.go) to demonstrate the usage of the generic API library. 
-After implementation sample could would be placed there - copy & paste it here and include a reference.
+## An example
+The sample apiserver provides an example admission plugin that makes meaningful use of the "standard" plugin initializer. 
+The admission plugin ensures that a resource name is not on the list of banned names.
+The source code of the plugin can be found [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission.go). 
 
+Having the plugin the next step is the regitration. [AdmissionOptions](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/options/admission.go)
```
Comment (sttts) [inline contributors/design-proposals/apiserver-build-in-admission-plugins.md]: registration

Comment (sttts) [inline contributors/design-proposals/apiserver-build-in-admission-plugins.md]: maybe comma after plugin?

File: contributors/design-proposals/apiserver-build-in-admission-plugins.md [hunk 5]
```diff
@@ -53,6 +53,28 @@ following series of steps:
...
+provides two important things. Firstly it exposes [a register](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/options/admission.go#L43) 
+under which all addmission plugins are registered. In fact that's exactly what the [Register](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission.go#L33)
```
Comment (sttts) [inline contributors/design-proposals/apiserver-build-in-admission-plugins.md]: In fact,

File: contributors/design-proposals/apiserver-build-in-admission-plugins.md [hunk 6]
```diff
@@ -53,6 +53,28 @@ following series of steps:
...
+## An example
+The sample apiserver provides an example admission plugin that makes meaningful use of the "standard" plugin initializer. 
+The admission plugin ensures that a resource name is not on the list of banned names.
+The source code of the plugin can be found [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission.go). 
 
+Having the plugin the next step is the regitration. [AdmissionOptions](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/options/admission.go)
+provides two important things. Firstly it exposes [a register](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/options/admission.go#L43) 
+under which all addmission plugins are registered. In fact that's exactly what the [Register](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission.go#L33) 
+method does from our example addmision plugin. It accepts a global registry as a parameter and then simply registers itself in that registry.
```
Comment (sttts) [inline contributors/design-proposals/apiserver-build-in-admission-plugins.md]: admission

File: contributors/design-proposals/apiserver-build-in-admission-plugins.md [hunk 7]
```diff
@@ -53,6 +53,28 @@ following series of steps:
...
+Secondly it adds admission chain to the server configuration via [ApplyTo](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/options/admission.go#L66) method.
```
Comment (sttts) [inline contributors/design-proposals/apiserver-build-in-admission-plugins.md]: Secondly,

Comment (sttts) [inline contributors/design-proposals/apiserver-build-in-admission-plugins.md]: Am not native speak, but maybe "adds a admission chain" ... "via a method".

File: contributors/design-proposals/apiserver-build-in-admission-plugins.md [hunk 8]
```diff
@@ -53,6 +53,28 @@ following series of steps:
...
+The method accepts optional parameter in the form of `pluginInitalizers`. This is useful when admission plugins need custom configuration that is not provided by the generic initializer.
```
Comment (sttts) [inline contributors/design-proposals/apiserver-build-in-admission-plugins.md]: parameters

File: contributors/design-proposals/apiserver-build-in-admission-plugins.md (hunk 3 above)
Comment (p0lyn0mial) [inline contributors/design-proposals/apiserver-build-in-admission-plugins.md]: sure, how can I change that to a fixed version ?

Comment (deads2k) [inline contributors/design-proposals/apiserver-build-in-admission-plugins.md]: https://github.com/kubernetes/kubernetes/blob/2f00e6d/staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission.go

**Comment by p0lyn0mial** [comment]
@sttts, @deads2k thanks for your comments, can we move this forward ?

**Pull request merged**